2_process.R

```diff
@@ -0,0 +1,48 @@
+source("2_process/src/pair_nhd_reaches.R")
+source("2_process/src/pair_nhd_catchments.R")
+source("2_process/src/create_GFv1_NHDv2_xwalk.R")
+source("2_process/src/write_data.R")
+source("2_process/src/write_ind_files.R")
+
+p2_targets_list <- list(
+  
+  # Pair PRMS segments (NHGFv1) with intersecting NHDPlusV2 reaches and contributing NHDPlusV2 catchments
+  tar_target(
+    p2_prms_nhdv2_xwalk,
+    create_GFv1_NHDv2_xwalk(prms_lines = p1_GFv1_reaches_sf,
+                            nhd_lines = p1_nhdv2reaches_sf,
+                            prms_hrus = p1_GFv1_catchments_sf,
+                            min_area_overlap = 0.5,
+                            drb_segs_spatial = drb_segs_spatial)
+  ),
+  
+  # Save GFv1-NHDv2 xwalk table 
+  tar_target(
+    p2_prms_nhdv2_xwalk_rds,
+    write_to_rds(p2_prms_nhdv2_xwalk,"2_process/out/GFv1_NHDv2_xwalk.rds"),
+    format = "file"
+  ),
+  
+  # Create and save indicator file
+  tar_target(
+    p2_data_summary_csv,
+    write_ind_files("2_process/log/GFv1_data_summary.csv",
+                    target_names = c("p1_GFv1_reaches_sf","p1_GFv1_catchments_sf",
+                                     "p1_nhdv2reaches_sf","p2_prms_nhdv2_xwalk","p2_prms_nhdv2_xwalk_rds")),
+    format = "file"
+  ),
```

Comment:
I'm running the pipeline for the first time and I noticed that this target built first. I'll check to see if it builds again after the other targets are run, or if it appears as outdated after the pipeline completes.

Reply:
I can confirm that this file is blank and the target does not appear as outdated

Reply:
well, it has headers:
<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/jsmith/AppData/Local/Temp/4/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/jsmith/AppData/Local/Temp/4/msohtmlclip1/01/clip_filelist.xml">

<!--table
	{mso-displayed-decimal-separator:"\.";
	mso-displayed-thousand-separator:"\,";}
@page
	{margin:.75in .7in .75in .7in;
	mso-header-margin:.3in;
	mso-footer-margin:.3in;}
tr
	{mso-height-source:auto;}
col
	{mso-width-source:auto;}
br
	{mso-data-placement:same-cell;}
td
	{padding-top:1px;
	padding-right:1px;
	padding-left:1px;
	mso-ignore:padding;
	color:black;
	font-size:11.0pt;
	font-weight:400;
	font-style:normal;
	text-decoration:none;
	font-family:Calibri, sans-serif;
	mso-font-charset:0;
	mso-number-format:General;
	text-align:general;
	vertical-align:bottom;
	border:none;
	mso-background-source:auto;
	mso-pattern:auto;
	mso-protection:locked visible;
	white-space:nowrap;
	mso-rotate:0;}
-->

</head>

<body link="#0563C1" vlink="#954F72">



tar_name | hash
-- | --




</body>

</html>

Reply:
Oh, this may have happened because your log files are in the repo, but the target objects are not. So, when I ran `tar_make`, the first detected change was that all the targets are now missing and it updated the log file. Definitely not something we accounted for with these files!

Reply:
It could be that the `target_names` need to somehow be specified unquoted to be tracked by targets. `tar_visnetwork` doesn't show these `target_names` as dependencies to the log file.

Reply:
Great catch! I think this was happening because the log files are in the repo before the targets get built (your point above). I added a cue to the log file targets so that they get built every time you run `tar_make()`. I think this solves the problem, but let me know if you see otherwise on your end.

Reply:
> It could be that the target_names need to somehow be specified unquoted to be tracked by targets.

The character string gets passed to a function we've defined to subset the `tar_meta()` table, so I don't think that's the crux of the issue here.

Reply:
I tried rebuilding from scratch and the `p2_data_summary_csv` target is still built first, resulting in a file with only headers. Looking at the target invalidation rules [here](https://rdrr.io/cran/targets/man/tar_cue.html), I wonder if committing the meta file and the object for `p2_data_summary_csv` would solve the problem. That way `p2_data_summary_csv` would only update if a dependency changed.

> The character string gets passed to a function

I was wondering if this properly tracks target dependencies when the target is only specified as a string. The dependency diagram does not show the targets contributing to `p2_data_summary_csv`. Looking into what [dependencies are tracked](https://books.ropensci.org/targets/targets.html#dependencies), I don't think strings would be tracked

```
Not all of the objects from tar_deps() actually register as dependencies. When it comes to detecting dependencies, targets only recognizes

Other targets (such as raw_data_file).
Functions and objects in the main environment. This environment is almost always the global environment of the R process that runs the target script file (default: _targets.R) so these dependencies are usually going to be the custom functions and objects you write yourself.
```

Reply:
I tried with `deparse(substitute(p1_GFv1_reaches_sf))` but that also didn't track the target dependency

Reply:
The problem here is that the target `p2_data_summary_csv` is building before any of the other data targets in the pipeline and so for fresh builds the call to `tar_meta()` was coming up blank and returning an empty data table where hashes should be. 

I'm not sure that this issue stems from the fact that the log files get committed since changing `tar_cue` to `always` had no effect on this behavior. Rather, as you point out above @jds485, I think we need to impose dependencies on the data summary target. This is hacky - so I'm interested in cleaner fixes - but for now I've opted to add an argument called `force_dep` to the function that creates the data summary table. As the name implies, `force_dep` just forces the dependency with an upstream target so that the targets get built in our desired order (i.e. build the metadata table _after_ building the crosswalk table). 

```
  # Create and save indicator file
  # argument force_dep must contain name of an upstream target to force dependencies
  # and build this target when a log file already exists
  tar_target(
    p2_data_summary_csv,
    write_ind_files("2_process/log/GFv1_data_summary.csv",
                    force_dep = p2_prms_nhdv2_xwalk_csv,
                    target_names = c("p1_GFv1_reaches_sf","p1_GFv1_catchments_sf","p1_nhdv2reaches_sf",
                                     "p2_prms_nhdv2_xwalk","p2_prms_nhdv2_xwalk_csv")),
    format = "file"),
```